### DISCUSSION TITLE
When I enable VoWiFi with a carrier that I have VoLTE enabled by default VoLTE gets unavailable

### DISCUSSION THREAD

**Discussion by ofir29200** [kyujin-cho/pixel-volte-patch#33] [Q&A] [answered]
Hi,
I am running the latest beta 13 (T2B3.230109.004) on my google pixel 6 pro
When I turn on VoWiFi, VoLTE gets unavailable in the ims service status:
![Screenshot_20230211-151521~2.png](https://user-images.githubusercontent.com/14427360/218260237-c7944900-a65c-47c7-9f2a-dbc4f8f80227.png)

Does someone know how this can be fixed?

**Comment by csaniag** [comment]
reset 'WiFi, mobile and bluetooth' under settings -> reset options

**Comment by ofir29200** [comment]
After reading https://github.com/kyujin-cho/pixel-volte-patch/issues/24 it sounds like it's the same thing, just on my pixel 6 pro

**Comment by arsradu** [comment]
Isn't that how it's supposed to be...? For as far as I know, you can't have both enabled at the same time. You will have VoLTE when Carrier Data is enabled and you are in a 4G/LTE area, and you will have VoWIFI if enabled and if you're connected to a WiFi network.

What would be the point of having both active at the same time? You can't place both a regular VoLTE call and VoWIFI call at the same time. 
It should be the same if you disable WiFi, but keep Mobile Data enabled. You will have VoLTE Available but VoWIFI Unavailable (cause you're not connected to WiFi).

So...unless I'm missing something, I think this is intended.

**Reply by ofir29200** [reply]
I am not saying that both need to be available, I am saying that the wifi calling is disabling the VoLTE correctly but not connecting the wifi calling itself

**Reply by eanavalentin** [reply]
> Isn't that how it's supposed to be...? For as far as I know, you can't have both enabled at the same time. You will have VoLTE when Carrier Data is enabled and you are in a 4G/LTE area, and you will have VoWIFI if enabled and if you're connected to a WiFi network.
> 
> What would be the point of having both active at the same time? You can't place both a regular VoLTE call and VoWIFI call at the same time. It should be the same if you disable WiFi, but keep Mobile Data enabled. You will have VoLTE Available but VoWIFI Unavailable (cause you're not connected to WiFi).
> 
> So...unless I'm missing something, I think this is intended.

Except that he has Not Registered for IMS and unavailable for both VoLTE/VoWiFi.

**Reply by arsradu** [reply]
> I am not saying that both need to be available, I am saying that the wifi calling is disabling the VoLTE correctly but not connecting the wifi calling itself

I see... What is the carrier? Is the service activated on carrier side...? It works perfectly fine in my case. So it doesn't seem to be a bug... Could be specific to this particular carrier though. Does it work on other phones (with support for both VoLTE and VoWIFI)?

**Reply by arsradu** [reply]
> Except that he has Not Registered for IMS and unavailable for both VoLTE/VoWiFi.

Yeah... Well, we don't have the whole picture (quite literally). :)) The only way it would show like this would be if you have no data connection at all. No mobile data, and no wifi.
But it's interesting that VoWIFI doesn't get enabled. I'm assuming it is already enabled in Pixel IMS app, and of course, connected to a WiFi network.
Other than that...I don't know... Maybe a Reset Network could do the trick. Or...a restart? Sometimes it doesn't work the first try. But it works afterwards.

**Reply by eanavalentin** [reply]
Not really, this is with both VoWiFi/VoLTE unchecked in the patch app. It just means the services aren't available.
Note that I have both data on and WiFi connected.
![Screenshot_20230212-133659.png](https://user-images.githubusercontent.com/10355724/218308620-cf281cde-2864-4d7d-b7f7-6e55c0d6a294.png)

**Reply by arsradu** [reply]
Oh, I see... So data is available, just not "patched" to be used by/for IMS services... Hmmm...then I don't know. If everything's correctly set up (and available) on both the carrier's side, and in the Pixel IMS app...then...I don't know. But...since this is not a general issue (I can't replicate it on my side), then it means it's probably the carrier, the OS, or...the IMS app is doing or not doing something for that particular case.

I think we might need more information here. It just got interesting. :))

**Reply by ofir29200** [reply]
@arsradu 
Well, my carrier is Partner Communications in Israel.
As for VoLTE, it's working when I am not connected to wifi (without the pixel ims) when I turn on pixel ims VoLTE goes unavailable, and wifi calling isn't turning on.
I have tested the wifi calling with an iPhone and it is enabled so it's not a carrier issue.

As for other users in my region that use this but with the regular android version, it does work for them (they even get the wifi icon)

**Reply by arsradu** [reply]
I see... Well, that narrows it down to two things: phone/Android version, or IMS app.

You could try resetting network settings. But keep in mind that will also erase all wifi passwords and BT pairings.

So, basically you only need VoWIFI/WiFi Calling support? VoLTE already works natively? If you only select VoWIFI in Pixel IMS app, does it work? 
Can you confirm the WiFi Calling option is available and enabled in Settings > SIMs > your carrier > Wi-Fi Calling?

**Reply by ofir29200** [reply]
First, yes, I only need VoWiFi, I already have VoLTE natively.

Secondly, I tried your suggestion to reset the network settings and it did not change anything, even when I selected the wifi first on settings. (Yes, it does show the wifi calling on the settings page)

**Reply by ofir29200** [reply]
Anything new?

**Reply by aronsky** [reply]
@ofir29200 I'm curious, how have you managed to get VoLTE working natively on Partner? I'm on 012 (same thing, really), and up until I found out about this app, I had no VoLTE on my Pixel 6.

**Reply by ofir29200** [reply]
> @ofir29200 I'm curious, how have you managed to get VoLTE working natively on Partner? I'm on 012 (same thing, really), and up until I found out about this app, I had no VoLTE on my Pixel 6.

I just updated to the beta version of Android 13 (I'm on T2B3.230109.009) and I called the technical support to enable VoLTE on my account, and it worked.

But from what I understand, 012 is not part of this yet

**Reply by aronsky** [reply]
Good to know. At least it's a step in the right direction.

**Reply by ofir29200** [reply]
> Good to know. At least it's a step in the right direction.

You could move your number to Partner (I think you won't even need to switch your sim card) and have VoLTE and wifi calling

**Comment by bige93** [comment]
I tried user agents from Telekom.de and T-Mobile US SIMs on my T-Mobile PL SIM - still no luck with VoWiFi.

**Comment by kszaq** [comment, marked as answer]
Try to add "ims" APN as described in https://github.com/kyujin-cho/pixel-volte-patch/issues/17#issuecomment-1435661999

Note that it may be invisible after adding - IMS is hidden by default - but may help in getting VoWiFi.

**Reply by ofir29200** [reply]
I tried it, but it doesn't work for me.
It still won't connect to wifi calling.

**Reply by bige93** [reply]
Worked for me with T-Mobile!
![Screenshot_20230218-224637](https://user-images.githubusercontent.com/66921212/219900914-0383198d-02cd-4a32-955b-19185caa8b41.png)
![Screenshot_20230218-224817](https://user-images.githubusercontent.com/66921212/219900953-20ca6dfe-9b99-44b3-ae7a-831c45f7d0f5.png)
Thank you so much @kszaq !

**Reply by ofir29200** [reply]
I tried it now and only after a reboot it worked!
Thank you all!

**Reply by ofir29200** [reply]
Now I have a new issue.
When I disconnect from wifi it won't connect to VoLTE, and when I connect back to the same wifi network it won't connect to wifi calling.
It will connect back only after a reboot (both the VoLTE and wifi calling)

**Reply by kszaq** [reply]
> Now I have a new issue.
> When I disconnect from wifi it won't connect to VoLTE, and when I connect back to the same wifi network it won't connect to wifi calling.
> It will connect back only after a reboot (both the VoLTE and wifi calling)

You can try to experiment with "bearer" setting for IMS APN. E.g. select both "Unspecified" and "LTE".

**Reply by ofir29200** [reply]
> > Now I have a new issue.
> > When I disconnect from wifi it won't connect to VoLTE, and when I connect back to the same wifi network it won't connect to wifi calling.
> > It will connect back only after a reboot (both the VoLTE and wifi calling)
> 
> You can try to experiment with "bearer" setting for IMS APN. E.g. select both "Unspecified" and "LTE".

Those are all of the options I have, what do they mean?

![Screenshot_20230219-181651.png](https://user-images.githubusercontent.com/14427360/219960527-534964f2-0cc2-4874-bd83-edaa6e61fd25.png)

![Screenshot_20230219-181656.png](https://user-images.githubusercontent.com/14427360/219960526-a7b0f938-0768-4cbe-88ff-ff6e68d2045f.png)

**Reply by kszaq** [reply]
> > > Now I have a new issue.
> > > When I disconnect from wifi it won't connect to VoLTE, and when I connect back to the same wifi network it won't connect to wifi calling.
> > > It will connect back only after a reboot (both the VoLTE and wifi calling)
> > 
> > You can try to experiment with "bearer" setting for IMS APN. E.g. select both "Unspecified" and "LTE".
> 
> Those are all of the options I have, what do they mean?
> 
> ![Screenshot_20230219-181651.png](https://user-images.githubusercontent.com/14427360/219960527-534964f2-0cc2-4874-bd83-edaa6e61fd25.png)
> 
> ![Screenshot_20230219-181656.png](https://user-images.githubusercontent.com/14427360/219960526-a7b0f938-0768-4cbe-88ff-ff6e68d2045f.png)
> 

You can select which type of network the APN applies to. But actually nevermind, selecting "Unspecified" should make it work for all network types. Perhaps there is some additional IMS setting your operator requires.

**Reply by kszaq** [reply]
One more thing: try to select IPv4/IPv6 for APN protocol and roaming protocol.

**Reply by bige93** [reply]
@ofir29200 Did this help? I had similar issues, but somehow now they're gone with no input from my side.

**Reply by ofir29200** [reply]
> @ofir29200 Did this help? I had similar issues, but somehow now they're gone with no input from my side. 

Well, sometimes it works, and sometimes it doesn't.
If I am connected to wifi, wifi calling works, but when I disconnect from wifi, sometimes it switches to VoLTE, and sometimes it won't switch to VoLTE and just disable the ims service until I restart my pixel (and i can't figure out why)

**Comment by ofir29200** [comment]
Hey everyone,
After I installed the new update and ran the app again, I got some of the same behavior as before, but it's different.
WiFi calling is only working after I reboot the phone (and only if I am connected to wifi before I reboot) and that's the same for VoLTE.
If I have WiFi calling working, then VoLTE won't work, and if I have VoLTE working, then WiFi calling won't work.

**Comment by ofir29200** [comment]
Hi everyone,
I have a discovery I want to share with you all.

While I have wifi calling working all the time (when i am connected to wifi) VoLTE will connect, but after a very long time (or a reboot).

The only way I found to make the pixel connect back to VoLTE at my choosing, was to ask someone with VoLTE already enabled and working to call me (and then for some reason it connects back to VoLTE and won't disconnect until I have wifi connected)

**Comment by Smooooochy** [comment]
Hi everyone, sorry for bumping a thread from 7 months ago.

I'm on Pixel 6a, connected to Partner (IL). Up until I installed Shizuku to enable VoWiFi, VoLTE worked perfect (natively).
Now that VoWiFi is finally enabled and working properly (after adding the IMS APN), I'm struggling with the same issue described above:

Every time I connect to my home wifi I get VoWiFi, but when I disconnect from it, LTE stops working (switching to H+ when making a call). I found out that restarting the device fixes it -- until I reconnect to my home Wifi and then it stops working again.

Searched the discussion board here and on the web and found nothing, while this thread is the closest I got so far.

@ofir29200 what are the chances you found an actual viable solution that doesn't require a restart (or a phone call from a friend :D)?
Anyone else?

Thanks!

**Reply by ofir29200** [reply]
Hey,
Now I'm on Android 14 (U1B2.230922.010) and it's working flawlessly.
I think it's something with the VoLTE itself like @arsradu said.

I suggest you contact the support line at partner to figure this out

**Reply by Smooooochy** [reply]
I'll be fully honest with you, Partner's support line is by far the last resort, imo. They're very unprofessional, and the moment you'll state you're using a Pixel phone they just get lost.

In any case, I'm on Android 14 as well; I'm assuming you're on the QPR beta?
I've opted in betas before, I don't mind trying again (if that's the case).

**Comment by arsradu** [comment]
Hey there.

So.....the issue is with VoLTE, for as far as I can tell. I don't see how Pixel IMS would cause this, to be honest. I'm on a Pixel 6, but not the same carrier or country, so....could be something specific in your case.  Or an Android bug. But I don't see why it would be a Pixel IMS issue. Neither caused, nor related to it...

I'm guessing you only enabled VoWIFI in Pixel IMS, right?

Anyway, you could also try one of the Beta builds of Android 14 QPR (now Beta 2.1 if I'm correct). I see there have been some fixes and changes made to the functionality of VoLTE and VOWIFI in the previously unsupported countries. I'm expecting with the final build of QPR (probably in December or January), everything will be fixed and working properly. But if you want to test the functionality, and see if it still occurs, you could do that, too...

The only problem is that you can't easily go back, in case it's not fixed. You can only mitigate the issues with Pixel IMS. But, if you're on the latest/"stable" build of Android 14, and you're willing to try, I would also try that, just in case it's an Android bug and might have been fixed in the meantime

Also, if you have a friend on the same network, willing to lend you his SIM card for a 5 minute test, you could also try that, and see if it makes a difference.

In my opinion (and I could be wrong), right now, it sounds more like an Android issue than a Pixel IMS issue.

**Reply by Smooooochy** [reply]
Hey! Thank you so much for replying so fast, with a thorough feedback.

I've opted-in betas before, I don't mind trying again. I'll take a look, and if it's sorted out I'll update!

Thanks!

**Comment by Smooooochy** [comment]
Ok!

Updated to QPR1 (U1B2.230922.010, from stable A14), and nothing, including "native" VoLTE support worked...

Had to remove and recreate the IMS APN, and after messing around a little bit with Shizuku and Pixel IMS (alongside several device restarts and IMS re-registrations), **seems that everything is now working properly!**

Switching from Wifi/LTE seems to keep IMS registered now, so it doesn't lose neither VoLTE or VoWIFI any longer.

Thanks for the tip fellas! (...and now back to nitpicking with beta bugs 😂)

**Reply by ofir29200** [reply]
I'm glad to hear that worked for you!

Enjoy your new operational VoLTE and VoWiFi!

**Reply by arsradu** [reply]
You're most welcome! Glad to hear it's working now! :)

**Reply by eliavsd** [reply]
@ofir29200 trying to achieve the same on Cellcom IL. Any tips for monkeying around with this?

I tried manually defining ims APN with a few different values, but so far VoWiFi doesn't work (it's activated on Cellcom's side) and VoLTE only works for outgoing calls.

**Reply by ofir29200** [reply]
> @ofir29200 trying to achieve the same on Cellcom IL. Any tips for monkeying around with this?
> 
> I tried manually defining ims APN with a few different values, but so far VoWiFi doesn't work (it's activated on Cellcom's side) and VoLTE only works for outgoing calls.

I already have VoLTE with partner, but I am still stuck with the VoWifi after the new update

I have enabled 5G with Network Signal Guru, so everything is good there
